webui/src/lib/components/repository/changes.jsx

```diff
@@ -89,65 +96,79 @@ export const TreeItem = ({ entry, repo, reference, leftDiffRefID, rightDiffRefID
-            }
-          </>
+        <TreeEntryRow key={entry.path + "entry-row"} entry={entry} dirExpanded={dirExpanded} relativeTo={relativeTo} depth={depth} onClick={() => setDirExpanded(!dirExpanded)} onRevert={onRevert} onNavigate={onNavigate} repo={repo} reference={reference} getMore={getMore}/>
+        {dirExpanded && results &&
+            results.map(child =>
+                (<TreeItem key={child.path + "-item"} entry={child} repo={repo} reference={reference} leftDiffRefID={leftDiffRefID} rightDiffRefID={rightDiffRefID} onRevert={onRevert} onNavigate={onNavigate}
+                           internalReferesh={internalRefresh} delimiter={delimiter} depth={depth + 1}
+                           relativeTo={entry.path} getMore={getMore}/>))}
+        {(!!nextPage || loading) &&
+            <TreeEntryPaginator path={entry.path} depth={depth} loading={loading} nextPage={nextPage}
+                                setAfterUpdated={setAfterUpdated}/>
+        }
+    </>
 }
 
-export const TreeEntryRow = ({ entry, showActions, relativeTo="", leaf=false, dirExpanded, depth=0, onClick, loading=false, onRevert, onNavigate}) => {
+export const TreeEntryRow = ({entry, relativeTo = "", leaf = false, dirExpanded, diffExpanded, depth = 0, onClick, loading = false, onRevert, onNavigate, onClickExpandDiff = null, getMore}) => {
+    const [showRevertConfirm, setShowRevertConfirm] = useState(false)
     let rowClass = 'tree-entry-row ' + diffType(entry);
     let pathSection = extractPathText(entry, relativeTo);
     let diffIndicator = diffIndicatorIcon(entry);
-    let actions = entryActions(showActions, entry, onRevert);
-
-    if (entry.path_type === "common_prefix"){
+    const [showSummary, setShowSummary] = useState(false);
+    if (entry.path_type === "common_prefix") {
         pathSection = <Link href={onNavigate(entry)}>{pathSection}</Link>
     }
+    const rowActions = []
+    if (onClickExpandDiff) {
+        rowActions.push(new RowAction(<FileDiffIcon/>, diffExpanded ? "Hide changes" : "Show changes", diffExpanded, onClickExpandDiff))
+    }
+    if (!leaf) {
+        rowActions.push(new RowAction(<GraphIcon/>, showSummary ? "Hide summary" : "Calculate change summary", showSummary, () => setShowSummary(!showSummary)))
+    }
+    rowActions.push(new RowAction(<HistoryIcon/>, "Revert changes", false, () => {setShowRevertConfirm(true)}))
     return (
-        <tr className={rowClass} >
-            <td className="diff-indicator">{diffIndicator}</td>
-            <td className="tree-path">
-                <span style={{marginLeft: depth * 20 + "px"}}>
+        <tr className={rowClass}>
+            <td className="pl-4 col-auto p-2">{diffIndicator}</td>
+            <td className="col-9 tree-path">
+                <span style={{marginLeft: (depth * 20) + "px"}}>
                     <span onClick={onClick}>
-                        {leaf
-                            ? entry.type === 'conflict'
```

Comment:
Where are you preserving this logic which means - don't show the content diff option in case of a conflict?

Reply:
oopsy

Reply:
Done